### PR TITLE
activate provisioner repositories

### PR DESCRIPTION
**Pull request by MaximilianMeister**
depends on https://github.com/crowbar/crowbar-core/pull/18

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1530,11 +1530,46 @@ EOF
...
     if ! validate_data_bags; then
         complain 68 "Validation error in default data bags. Aborting."
     fi
 }
 
+function activate_one_provisioner_repository()
+{
+    local platform=$1
+    local repo=$2
+    curl --digest -u crowbar:crowbar -X POST -d "platform=$platform" -d "repo=$repo" http://$adminip/utils/repositories/activate
```
Comment (tboerger) [inline scripts/qa_crowbarsetup.sh]: Hum, the other curl commands just request `http://localhost:3000`

**Comment by bmwiedemann** [comment]
:+1: in the hope that the code will be gone again in 2 months

**Comment by MaximilianMeister** [comment]
commented out the complain until https://github.com/crowbar/crowbar-core/pull/18 is ready in the crowbar-core package

**Comment by vuntz** [comment]
This should be somehow done in install-suse-cloud, imho. We can't expect customers to go on the page to deploy nodes.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1524,11 +1524,49 @@ EOF
...
+        esac
+
+        if [ "$hacloud" == 1 ] ; then
+            activate_one_provisioner_repository "suse-12.1" "SLE12-SP1-HA-Pool"
+            activate_one_provisioner_repository "suse-12.1" "SLE12-SP1-HA-Updates"
+        fi
+
+        if [ -n "$deployceph" ] ; then
+            activate_one_provisioner_repository "suse-12.0" "SUSE-Enterprise-Storage-2.0-Pool"
+            activate_one_provisioner_repository "suse-12.0" "SUSE-Enterprise-Storage-2.0-Updates"
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: s/2.0/2/g

**Comment by MaximilianMeister** [comment]
> This should be somehow done in install-suse-cloud, imho. We can't expect customers to go on the page to deploy nodes.

@vuntz sorry then it was my misunderstanding, because if we wouldn't expect the customer to activate them explicitly by hand, why would we need a checkbox? i thought the design said it had to be explicit, and the trello card has a documentation task to file a bug.

**Comment by vuntz** [comment]
@MaximilianMeister it should be automatic at install time, and requiring an explicit action after that.

The rationale is that, by default, after installation, you want something that works. Then the user can choose to disable some repos, or add some new repos and enable them.

**Comment by MaximilianMeister** [comment]
@vuntz okay, then we have some default repos, which would be SLES-12-SP1-Pool and SLES12-SP1-Updates which should be activated in install-chef-suse.sh right?

**Comment by vuntz** [comment]
@MaximilianMeister all repos that are available should be activated by default in install-chef-suse.sh.

**Comment by MaximilianMeister** [comment]
@vuntz https://github.com/crowbar/crowbar/pull/2172

**Comment by MaximilianMeister** [comment]
closing in favor of https://github.com/crowbar/crowbar/pull/2172

**Pull request closed without merging**